wsgi.py

```diff
@@ -6,255 +6,348 @@
             return Response(
-                response=json.dumps({"msg": "project deleted (" + project_uuid +")" }),
+                response=json.dumps({"msg": "project deleted (" + project_uuid + ")"}),
                 status=200,
-                mimetype='application/json'
-               )
+                mimetype="application/json",
+            )
         return Response(
-            response=json.dumps({"msg": "project unabled to be deleted (" + project_uuid +")" }),
+            response=json.dumps(
+                {"msg": "project unabled to be deleted (" + project_uuid + ")"}
+            ),
             status=400,
-            mimetype='application/json'
-            )
+            mimetype="application/json",
+        )
     return Response(
-        response=json.dumps({"msg": "unable to delete, project does not exist(" + project_uuid +")" }),
+        response=json.dumps(
+            {"msg": "unable to delete, project does not exist(" + project_uuid + ")"}
+        ),
         status=400,
-        mimetype='application/json'
-        )
+        mimetype="application/json",
+    )
 
-@application.route("/users/<user_name>", methods=['GET'])
-def get_moc_user(user_name, full_name=None, id_provider="sso_auth", id_user=None):
-    (token, openshift_url) = get_token_and_url()
-    r=None
-    if(exists_openshift_user(token, openshift_url, user_name)):
+
+@application.route("/users/<user_name>", methods=["GET"])
+@auth.login_required
+def get_moc_user(user_name, full_name=None, id_provider="moc-sso", id_user=None):
+    shift = get_openshift()
+    r = None
+    if shift.user_exists(user_name):
         return Response(
             response=json.dumps({"msg": "user (" + user_name + ") exists"}),
             status=200,
-            mimetype='application/json'
-            )
+            mimetype="application/json",
+        )
     return Response(
-            response=json.dumps({"msg": "user (" + user_name + ") does not exist"}),
-            status=400,
-            mimetype='application/json'
-            )       
-
-@application.route("/users/<user_name>", methods=['PUT'])
-def create_moc_user(user_name, full_name=None, id_provider="sso_auth", id_user=None):
-    token, openshift_url = get_token_and_url()
-
-    # A user in OpenShift is composed of 3 parts: user, identity, and identitymapping.
-    user_exists = exists_openshift_user(token, openshift_url, user_name)
-    if not user_exists:
-        r = create_openshift_user(token, openshift_url, user_name, full_name)
-
-        if r.status_code not in [200, 201]:
+        response=json.dumps({"msg": "user (" + user_name + ") does not exist"}),
+        status=400,
+        mimetype="application/json",
+    )
+
+
+@application.route("/users/<user_name>", methods=["PUT"])
+@auth.login_required
+def create_moc_user(user_name, full_name=None, id_provider="moc-sso", id_user=None):
+    shift = get_openshift()
+    r = None
+    # full name in payload
+    user_exists = 0x00
```

Comment:
Please don't use bitfields like this. Even if you were writing in C, you would be using named constants rather than bare integers. Code like this doesn't give me any idea what's going on:

```
if user_exists == 7:
```

I'm not sure you need to track the individual state of each operation here. Given how you're using the value of `user_exists`, you could just have it be a single boolean variable named e.g. `i_created_something`. You start with it set to `False`, and then as you move through the function you set it to `True` if you need to create a user/identity/mapping/etc.

Then your conditional becomes:

```
if i_created_something:
```

Which is I think much easier to undestand.